rmgpy/statmech/torsion.pyx

```diff
@@ -526,32 +526,30 @@ cdef class HinderedRotor(Torsion):
         numterms = 6
         cdef bint negative_barrier
         negative_barrier = True
-        # numterms is actually half the number of terms. It is called numterms 
+        # numterms is actually half the number of terms. It is called numterms
         # because it is the number of terms of either the cosine or sine fit
 
         maxterms = np.floor(len(angle) / 3.0)
         while negative_barrier and numterms < maxterms:
```

Comment:
Can you change the `<` to `<=`?  Since we don't increment the numterms until after trying the fit and seeing if there is a negative_barrier, we never try maxterms.  For example, if your maxterms is 8 and you get negative barrier for 6 terms, since 8 is not less than 8, we won't try the fit with 8 terms.

Reply:
Yes, you are right.